src/stream/src/executor/aggregation/agg_impl/foldable.rs

```diff
@@ -649,4 +657,354 @@ mod tests {
         .unwrap();
         assert_eq!(agg.get_output().unwrap().unwrap().as_int64(), &100);
     }
+
+    fn gen_rand_bitmap(num_bits: usize, count_ones: usize) -> Bitmap {
+        let mut builder = BitmapBuilder::zeroed(num_bits);
+        let mut range = (0..num_bits).collect_vec();
+        range.shuffle(&mut rand::thread_rng());
+        let shuffled = range.into_iter().collect_vec();
+        for item in shuffled.iter().take(count_ones) {
+            builder.set(*item, true);
+        }
+        builder.finish()
+    }
+
+    fn bench_i64(
+        b: &mut Bencher,
+        mut agg: Box<dyn StreamingAggImpl>,
+        agg_desc: &str,
+        chunk_size: usize,
+        vis_rate: f64,
+        iter_count: usize,
+        append_only: bool,
+    ) {
+        println!(
+            "benching {} agg, chunk_size {}, vis_rate {}, iter_count {}",
+            agg_desc, chunk_size, vis_rate, iter_count
+        );
+        let mut rng = rand::thread_rng();
+        let mut ops: Vec<Op> = vec![];
+        let mut data_builder = I64ArrayBuilder::new(chunk_size);
+        let mut cur_data: Vec<i64> = vec![];
+        let bitmap = if vis_rate < 1.0 {
+            Some(gen_rand_bitmap(
+                chunk_size,
+                (chunk_size as f64 * vis_rate) as usize,
+            ))
+        } else {
+            None
+        };
+
+        if let Some(bitmap) = bitmap.as_ref() {
+            for i in 0..chunk_size {
+                // SAFETY(value_at_unchecked): the idx is always in bound.
+                unsafe {
+                    if bitmap.is_set_unchecked(i) {
+                        let op = if append_only || cur_data.is_empty() || rng.gen() {
+                            Op::Insert
+                        } else {
+                            Op::Delete
+                        };
```

Comment:
Can you refactor this out into its own function? Seems like logic is duplicated below.